src/azure-cli/azure/cli/command_modules/storage/oauth_token_util.py

```diff
@@ -1,48 +0,0 @@
-# --------------------------------------------------------------------------------------------
-# Copyright (c) Microsoft Corporation. All rights reserved.
-# Licensed under the MIT License. See License.txt in the project root for license information.
-# --------------------------------------------------------------------------------------------
-
-import threading
-
-
-class TokenUpdater(object):
-    """
-    This class updates a given token_credential periodically using the provided callback function.
-    It shows one way of making sure the credential does not become expired.
-    """
-    def __init__(self, token_credential, cli_ctx):
-        self.token_credential = token_credential
-        self.cli_ctx = cli_ctx
-
-        # the timer needs to be protected, as later on it is possible that one thread is setting a new timer and
-        # another thread is trying to cancel the timer
-        self.lock = threading.Lock()
-        self.timer_callback()
-
-    def timer_callback(self):
-        # call to get a new token and set a timer
-        from azure.cli.core._profile import Profile
-        from datetime import datetime
-        # should give back token that is valid for at least 5 mins
-        token = Profile(cli_ctx=self.cli_ctx).get_raw_token(resource="https://storage.azure.com")[0][2]
-        try:
-            self.token_credential.token = token['accessToken']
```

Comment:
It seems the token is accessed via `token_credential.token`. `AdalAuthentication` doesn't have this field, but only has `get_token`. Won't this break track 1 SDK?

Reply:
actually what we need is signed session in track 1. `token` is used in signed session in the updater.

Reply:
Aha, I see: https://github.com/Azure/azure-multiapi-storage-python/blob/726b1b0cf3c61aae52fc410451f968cbc4e675bc/azure/multiapi/storage/v2018_11_09/common/tokencredential.py#L25

```py
    def signed_session(self, session=None):
        """
        Sign requests session with the token. This method is called every time a request is going on the wire.
        The user is responsible for updating the token with the preferred tool/SDK.
        In general there are two options:
            - override this method to update the token in a preferred way and set Authorization header on session
            - not override this method, and have a timer that triggers periodically to update the token on this class
        The second option is recommended as it tends to be more performance-friendly.
        :param session: The session to configure for authentication
        :type session: requests.Session
        :rtype: requests.Session
        """
        session = session or requests.Session()
        session.headers['Authorization'] = "Bearer {}".format(self.token)

        return session

    def token(self, new_value):
        """
        :param new_value: new value to be set as the token.
        """
        self.token = new_value
```